### PR TITLE
Fix missing notification events issue

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
Fixes https://github.com/wso2/product-apim/issues/8340

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -2291,6 +2292,12 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
             apiMgtDAO.makeKeysForwardCompatible(new ApiTypeWrapper(api), oldVersion
             );
         }
+        int apiId = apiMgtDAO.getAPIID(api.getId(), null);
+        APIEvent apiEvent = new APIEvent(UUID.randomUUID().toString(), System.currentTimeMillis(),
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2296]: don't we hv to send subscription related events?

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: resolved

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -2288,8 +2289,7 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
         Collections.sort(sortedAPIs, comparator);
         for (int i = sortedAPIs.size() - 1; i >= 0; i--) {
             String oldVersion = sortedAPIs.get(i).getId().getVersion();
-            apiMgtDAO.makeKeysForwardCompatible(new ApiTypeWrapper(api), oldVersion
-            );
+            apiMgtDAO.makeKeysForwardCompatible(new ApiTypeWrapper(api), oldVersion, tenantId);
         }
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2293]: I don't think you have to pass the tenant Id , you could get the tenant id from the API provider using APIUtil.getTenantId(APIUtil.replaceEmailDomainBack(api.getId().getProviderName())

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:2293]: resolved

**Pull request merged**